scripts/ec2_stage_arm.sh

```diff
@@ -162,6 +161,23 @@ echo "${userdata}"
 echo "======================================================"
 echo
 
+# Install jq
+add-apt-repository ppa:eugenesan/ppa
+apt-get update
+apt-get install jq
+
+# Fetch the Arm AMI Id
+AMI_ID=$(aws ec2 describe-images \
```

Comment:
is this guaranteed to return one result?

Reply:
This does not guarantee one result but if the account has access to an Arm instance this should almost always return at least one result.

Reply:
I would not recommend doing this.  There are very few guarantees from the output here, including which specific account owns the image (a number of accounts are considered to be "amazon" when querying by owner).

The recommended way to retrieve AMI IDs for Amazon Linux is via SSM: https://aws.amazon.com/blogs/compute/query-for-the-latest-amazon-linux-ami-ids-using-aws-systems-manager-parameter-store/

Reply:
Can you make a todo note to use SSM when they are available?

Reply:
1. If there is no ID return, will ec2 run-instances fail and error will be thrown so that people who run the script can know what happened? Just want to double check we are able to debug if error happens.
2. Maybe add a todo about changing to SSM parameter store once it is available?

Reply:
I would lean toward leaving this hard-coded until the SSM parameters are available.  The approach here does not give you guaranteed, predictable results.

Reply:
We have had this pattern break in past situations. I don’t want us to do this.